corehq/apps/api/odata/views.py

```diff
@@ -22,8 +23,19 @@
 ], name='dispatch')
 
 
+class BaseODataView(View):
+
+    def dispatch(self, request, *args, **kwargs):
+        # domain = kwargs.get('domain')
+        # print(domain)
+        # print(request)
+        if not user_can_view_odata_feed(request.domain, request.couch_user):
+            raise Http404()
```

Comment:
I think this should instead be added to https://github.com/dimagi/commcare-hq/blob/573738a0cca7a229078e6f5b231d312d67ff137a/corehq/apps/api/odata/views.py#L18-L22

Reply:
`user_can_view_odata_feed` isn't a decorator. I thought this was easier and simpler than creating a special decorator just to be used once that would be included in the `odata_auth` list...